src/java.base/share/classes/java/util/ResourceBundle.java

```diff
@@ -3722,8 +3654,8 @@ private static String toPackageName(String bundleName) {
 
     }
 
-    private static final boolean TRACE_ON = Boolean.parseBoolean(
-        GetPropertyAction.privilegedGetProperty("resource.bundle.debug", "false"));
+    private static final boolean TRACE_ON = Boolean.getBoolean(
```

Comment:
This update seems to have broken the tracing feature of ResourceBundle.

The previous code called `GetPropertyAction::privilegedGetProperty` to get the system property "resource.bundle.debug" with a default of "false".  It then used `Boolean::parseBoolean` to compare it to "true", ignoring case.

The new code  uses `System::getProperty` to get the same property, then calls `Boolean::getBoolean` which calls `System::getProperty` to get the value of the property which name is either "false" or the result of looking up "resource.bundle.debug".

Seems like we could simply use `Boolean.getBoolean("resource.bundle.debug")` instead?

Reply:
> Seems like we could simply use `Boolean.getBoolean("resource.bundle.debug")` instead?

Naoto may know the history on this property. It may have been introduced for debugging when working on the RB implementation or maybe it was introduced to allow developers to debug, not sure. If the latter then it's important to preserve long standing behavior. If the former, and it was never documented, there is a lot more flexibility to change.

Reply:
> If the former, and it was never documented, there is a lot more flexibility to change.

Fair enough. However, this SM change seems to accidentally have introduced a bug where the system property is read twice, first calling System.getProperty to look up the name of the system property which is then looked up by  Boolean.getBoolean. That behavior just seems broken. 

@RogerRiggs may have intended to use `Boolean::parseBoolean` instead:

`Boolean.parseBoolean(System.getProperty("resource.bundle.debug", "false"))`

which is equivalent to:

`Boolean.getBoolean("resource.bundle.debug", "false")`

Reply:
Filed https://bugs.openjdk.org/browse/JDK-8345818.